### PR TITLE
Fix for bug in PHP FPM through both ProxyPassMatch and SetHandler

### PR DESCRIPTION
**Pull request by Caffe1neAdd1ct**
where the global variable SERVER SCRIPT_NAME isn't set correctly on apache 2.4.

### CODE REVIEW

**Comment by Caffe1neAdd1ct** [comment]
Workaround for a Bug in PHP which affects only Apache2 running with PHP-FPM: https://bugs.php.net/bug.php?id=65641

This might replace: #17491

Potential Bug References:

#22411
#21871
#7719

Backport request for RHEL 7: https://bugzilla.redhat.com/show_bug.cgi?id=1248077

@wpoely86
@RealRancor

**Comment by Caffe1neAdd1ct** [comment]
Manual tests i'm going to run through :

- [x]  Test webapp with apache ProxyPassMatch FPM
- [x]  Test webapp with apache SetHandler FPM
- [ ] Test webapp with apache mod_php
- [ ] Test webapp with nginx fpm setup
- [x]  Test sync client with apache ProxyPassMatch FPM
- [x]  Test sync client with apache SetHandler FPM
- [ ] Test sync client with apache mod_php
- [ ] Test sync client with nginx fpm setup
- [ ] Test /token/generate url still works
- [ ] Test /ocs/v1.php/ urls still works
- [ ] Test /ocs/v2.php/ urls still works

**Comment by Caffe1neAdd1ct** [comment]
@wpoely86 Apache config on my servers:

```
<VirtualHost *:80>

    ServerName owncloud.local
    ServerAdmin support@local
    DocumentRoot /usr/share/webapps/owncloud/

    SetEnvIf Authorization "(.*)" HTTP_AUTHORIZATION=$1
    SetEnv HOME /usr/share/webapps/owncloud
    SetEnv HTTP_HOME /usr/share/webapps/owncloud
    
    DirectoryIndex index.php
    ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi://127.0.0.1:9001/usr/share/webapps/owncloud/$1

    <Directory "/usr/share/webapps/owncloud">
        AllowOverride All
        Require all granted
    </Directory>

    ErrorLog "/var/log/httpd/owncloud.error.log"
    CustomLog "/var/log/httpd/owncloud.access.log" combined

</VirtualHost>
```

or via SetHandler

```
<VirtualHost *:80>

    ServerName owncloud.local
    ServerAdmin support@local
    DocumentRoot /usr/share/webapps/owncloud/

    SetEnvIf Authorization "(.*)" HTTP_AUTHORIZATION=$1
    SetEnv HOME /usr/share/webapps/owncloud
    SetEnv HTTP_HOME /usr/share/webapps/owncloud
    
    DirectoryIndex index.php

    <FilesMatch \.php$>
        SetHandler "proxy:fcgi://127.0.0.1:9001"
    </FilesMatch>

    <Directory "/usr/share/webapps/owncloud">
        AllowOverride All
        Require all granted
    </Directory>

    ErrorLog "/var/log/httpd/owncloud.error.log"
    CustomLog "/var/log/httpd/owncloud.access.log" combined

</VirtualHost>
```

**Comment by Caffe1neAdd1ct** [comment]
This pull request replaces https://github.com/owncloud/core/pull/25747 as my attempted rebase to squash commits used a localhost committer and rewriting history didn't go well (some learning to do there).

**Comment by Caffe1neAdd1ct** [comment]
@DeepDiver1975 1 commit as requested and all tests passing, could you add the relevant tags on the new PR?

Added some new tests as well to cover `/ocs/` and `/token/generate` as they use a slightly different uri format and need compatibility to work across the different setups.

**Comment by Caffe1neAdd1ct** [comment]
Are we able to get this new PR tagged for 9.2 as the issue it resolves is affecting a range of setups.

**Comment by PVince81** [comment]
9.2 should be fine, should give enough time for all the tests to be done

**Comment by Caffe1neAdd1ct** [comment]
:+1: shout if i can help with any of the testing

File: tests/lib/AppFramework/Http/RequestTest.php [hunk 1]
```diff
@@ -1222,12 +1228,14 @@ public function genericPathInfoProvider() {
 			['/index.php/apps/files/', 'index.php', '/apps/files/'],
 			['/index.php/apps/files/../&amp;/&?someQueryParameter=QueryParam', 'index.php', '/apps/files/../&amp;/&'],
 			['/remote.php/漢字編碼方法 / 汉字编码方法', 'remote.php', '/漢字編碼方法 / 汉字编码方法'],
-			['///removeTrailin//gSlashes///', 'remote.php', '/removeTrailin/gSlashes/'],
+			['/remote.php/removeTrailin//gSlashes///', 'remote.php', '/removeTrailin/gSlashes/'],
+            ['/token/generate', '/index.php', '/token/generate'],
+            ['/ocs/v2.php/cloud/users/user0', '/ocs/v2.php', '/cloud/users/user0'],
```
Comment (PVince81) [inline tests/lib/AppFramework/Http/RequestTest.php]: some indent issue here, please use tabs

Comment (Caffe1neAdd1ct) [inline tests/lib/AppFramework/Http/RequestTest.php]: No prob, I'll reconfigure netbeans for this project and fix the commit

**Comment by Caffe1neAdd1ct** [comment]
Spaces/Tabs issues fixed, plus hopefully resolved any introduced issues in Scrutinizer.

Tests are passing locally for BeHat and PHPUnit(sqlite) so waiting on full CI builds now.

**Comment by Caffe1neAdd1ct** [comment]
Jenkins build status:

```
Could not update commit status. Message: GitHub API rate limit exceeded

Finished: SUCCESS
```

**Comment by Caffe1neAdd1ct** [comment]
@DeepDiver1975 any feedback on why this can't be merged?

**Pull request closed without merging**